packages/solid-urql/src/createQuery.ts

```diff
@@ -0,0 +1,337 @@
+ * for the query, and changes when your input `args` change.
+ *
+ * Additionally, if the `suspense` option is enabled on the `Client`,
+ * the `createQuery` hook will suspend instead of indicating that it’s
+ * waiting for a result via {@link CreateQueryState.fetching}.
+ *
+ * @example
+ * ```tsx
+ * import { gql, createQuery } from '@urql/solid';
+ *
+ * const TodosQuery = gql`
+ *   query { todos { id, title } }
+ * `;
+ *
+ * const Todos = () => {
+ *   const [result, reExecuteQuery] = createQuery({
+ *     query: TodosQuery,
+ *   });
+ *   // ...
+ * };
+ * ```
+ */
+export const createQuery = <
+  Data = any,
+  Variables extends AnyVariables = AnyVariables,
+>(
+  args: CreateQueryArgs<Data, Variables>
+): CreateQueryResult => {
+  const client = useClient();
+  const getContext = asAccessor(args.context);
+  const getPause = asAccessor(args.pause);
+  const getRequestPolicy = asAccessor(args.requestPolicy);
+  const getVariables = asAccessor(args.variables);
+
+  const [source, setSource] = createSignal<
+    Source<OperationResult<Data, Variables>> | undefined
+  >(undefined, { equals: false });
+
+  // Combine suspense param coming from context and client with context being priority
+  const isSuspense = createMemo(() => {
+    const ctx = getContext();
+    if (ctx !== undefined && ctx.suspense !== undefined) {
+      return ctx.suspense;
+    }
+
+    return client.suspense;
+  });
+
+  const request = createRequest(args.query, getVariables() as any);
+  const context: Partial<OperationContext> = {
+    requestPolicy: getRequestPolicy(),
+    ...getContext(),
+  };
+  const operation = client.createRequestOperation('query', request, context);
+  const initialResult: CreateQueryState<Data, Variables> = {
+    operation: operation,
+    fetching: false,
+    data: undefined,
+    error: undefined,
+    extensions: undefined,
+    hasNext: false,
+    stale: false,
+  };
+
+  const [result, setResult] =
+    createStore<CreateQueryState<Data, Variables>>(initialResult);
+
+  createComputed(() => {
+    if (getPause() === true) {
+      setSource(undefined);
+      return;
+    }
+
+    const request = createRequest(args.query, getVariables() as any);
+    const context: Partial<OperationContext> = {
+      requestPolicy: getRequestPolicy(),
+      ...getContext(),
+    };
+
+    setSource(() => client.executeQuery(request, context));
+  });
+
+  createComputed(() => {
+    const s = source();
+    if (s === undefined) {
+      setResult(
+        produce(draft => {
+          draft.fetching = false;
+          draft.stale = false;
+        })
+      );
+
+      return;
+    }
+
+    setResult(
+      produce(draft => {
+        draft.fetching = true;
+        draft.stale = false;
+      })
+    );
+
+    onCleanup(
+      pipe(
+        s,
+        onEnd(() => {
+          setResult(
+            produce(draft => {
+              draft.fetching = false;
+              draft.stale = false;
+            })
+          );
+        }),
```

Comment:
Small question: Are we sure this triggers _before_ the new effect is set up, so we don't get `fetching: false` after the new effect started?

Just double checking that cleanups are called before the next source is set up.

Reply:
From what I see it is always called after dependencies (only dependency is `source` signal) change